### DISCUSSION TITLE
Leniency

### DISCUSSION THREAD

**Discussion by charlesdecourval** [nomemory/mapneat#5] [Q&A]
Hi,
how do I make the lib more lenient and not fail on a known  json path that is missing ?
So , if in the code I'm expecting $.company.location.address  and the address part is not in the json that I'm processing, how can I just ignore this and not fail ?
tks

**Comment by nomemory** [comment]
@charlesdecourval 

That's not supported at the DSL level, but I believe it's a good idea that can be implemented in a future release.

For the moment the only way to do it, it's like this:

```kotlin
@Test
    fun `Assignments are working correctly with inner JSONs and lambdas`() {
        testFromDirectory("assign/jsoninjson") {
            "author" /= json {
                "fullName" /= {
                    sourceCtx().read("$.books[1].author")
                }
                "firstName" /= { targetCtx().read<String>("$.fullName").split(" ")[0] }
                "lastName" /= { targetCtx().read<String>("$.fullName").split(" ")[1] }
                - "fullName"
                "book" /= {
                    try {
                        sourceCtx().read("$.books[1].title1")
                    } catch (e: PathNotFoundException) {
                        ""
                    }
                }
                "address" /= json {
                    "country" /= "UK"
                }
            }
        }.doTest()
    }
```

You need to catch the exception `PathNotFoundException`:

```kotlin
"book" /= {
                    try {
                        sourceCtx().read("$.books[1].title1")
                    } catch (e: PathNotFoundException) {
                        ""
                    }
                }
```

Do you have any suggestions on how to implement this, from a design perspective?

Andrei

**Comment by charlesdecourval** [comment]
After using the lib for most of the year, here's what we came up with, with success

```
fun MapNeat.assignHide(dest: String, src: String, log: Logger) {
  try {
    dest *= src
  } catch (exception: Exception) {
    log.warn(TransformerConstant.ERROR_PARSING + exception.message)
  }
}
```

But what would be nice would be an operator of the sort
/=?     that would hide the exception
or
*=?     that would do the same.

but over all, love the lib !

**Reply by nomemory** [reply]
I've just saw your message. I will take this in consideration for the next release. Sorry for taking so long to respond, but I was quite busy with other assignments.
